lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2401,6 +2401,13 @@ public function clear($entityName = null)
                     $this->doDetach($entity, $visited, false);
                 }
             }
+
+            foreach ($this->entityInsertions as $hash => $entity) {
```

Comment:
I could ofcourse, but those are not my contributions. 
Would you like me to alter that ?